### PR TITLE
FISH-6972 Instrument JAX-RS Clients and MP REST Clients in OTEL-Enabled Application

### PR DESCRIPTION
**Pull request by arieki**
## Description

Refactor JAX-RS Client intrumentation by enabling OpenTelemetry

### Blockers

N/A

### New tests

@pdudits's PR https://github.com/payara/Payara/pull/6217 creates OTel testing on payara-sample

### Testing Performed

+ Build and run Payara server
+ MP-Telemetry TCK
+ MP-OpenTracing TCK

### Testing Environment

Windows 11, OpenJDK 11, Maven 3.8.4

## Documentation

None

## Notes for Reviewers

None

### CODE REVIEW

**Comment by arieki** [comment]
jenkins test please

**Comment by arieki** [comment]
jenkins test please

**Comment by pdudits** [comment]
Jenkins test please

**Comment by arieki** [comment]
jenkins test please

**Review by pdudits — APPROVED** [review]
Added access to span in propagation helper, I don't get any more NPEs from Request Tracing Service

File: appserver/payara-appserver-modules/microprofile/telemetry/src/main/java/fish/payara/microprofile/telemetry/tracing/jaxrs/client/JaxrsClientRequestTelemetryFilter.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                    .setSpanKind(SpanKind.CLIENT);
+
+            // Get the propagated span context from the request if present
+            // This is required to account for asynchronous client requests
+            var parentSpanContext = (Context) requestContext.getProperty(PropagationHeaders.TELEMETRY_PROPAGATED_SPANCONTEXT);
+            if (parentSpanContext != null) {
+                spanBuilder.setParent(parentSpanContext);
+            } else {
+                parentSpanContext = Context.current();
+                if (parentSpanContext != null) {
```
[the remark this answers is not in the document]
Comment (arieki) [reply appserver/payara-appserver-modules/microprofile/telemetry/src/main/java/fish/payara/microprofile/telemetry/tracing/jaxrs/client/JaxrsClientRequestTelemetryFilter.java:131]: you are right, I will remove null checker

**Comment by arieki** [comment]
jenkins test please

**Comment by pdudits** [comment]
jenkins test please

**Pull request merged**